diff --git a/pykolab/auth/ldap/__init__.py b/pykolab/auth/ldap/__init__.py
index 4b5f876..7ba3504 100644
--- a/pykolab/auth/ldap/__init__.py
+++ b/pykolab/auth/ldap/__init__.py
@@ -325,6 +325,11 @@ class LDAP(Base):
                     self._disconnect()
                     return False
 
+                # store current unix time in last login
+                self.bind = False
+                if self.config_get('setlastlogin') == "True" and not "uid=cyrus-admin" in entry_dn:
+                    self.set_entry_attribute(entry_dn, "tbitsKolabLastLogin", str(int(time.time())))
+
                 try:
                     auth_cache.set_entry(_filter, entry_dn)
                 except Exception as errmsg:
@@ -369,6 +374,11 @@ class LDAP(Base):
                     self._disconnect()
                     return False
 
+                # store current unix time in last login
+                self.bind = False
+                if self.config_get('setlastlogin') == "True" and not "uid=cyrus-admin" in entry_dn:
+                    self.set_entry_attribute(entry_dn, "tbitsKolabLastLogin", str(int(time.time())))
+
             except ldap.NO_SUCH_OBJECT as errmsg:
                 log.debug(
                     _l("Error occured, there is no such object: %r") % (
